### ISSUE TITLE
Add support for Wide Color Gamut images

### ISSUE THREAD

**Issue by ay8s** [pinterest/PINRemoteImage#301]
It looks like PINRemoteImage is using UIImagePNGRepresentation & UIImageJPEGRepresentation which isn't wide color aware. It would need to start using the new UIGraphicsImageRenderer introduced in iOS 10. 

Instagram shared some findings when adding support for this in their app here... https://engineering.instagram.com/bringing-wide-color-to-instagram-5a5481802d7d#.24vy6avtt

**Comment by DreamingInBinary** [comment]
Hey @garrettmoon -

I took a look at this one this morning. Wanted to run some thoughts across you. From what I can see, to make images wide color aware and show up on Display P3 iOS/macOS devices, two different things need to be considered: when images are imported and when they are consumed/exported.

In regards to PINRemoteImage then, I think that means we'd need to:

1) Edit any Core Graphics manipulation to instead use the new `UIImageGRaphicsRenderer`

So anywhere were there are things like this going on:
```
UIGraphicsBeginImageContextWithOptions(…)
//Drawing stuff/crops/resize/etc
UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
UIGraphicsEndImageContext();
```
..we'd want to instead refactor those to be something like this:
```
UIGraphicsImageRendererFormat *format =  [UIGraphicsImageRendererFormat new];
format.prefersExtendedRange = YES;
UIGraphicsImageRenderer *renderer = [[UIGraphicsImageRenderer alloc] initWithSize:size format:format];
//Do any manipulation
```

This is also an iOS 10 API, so any iOS 9 targets would simply fall back into the old Core Graphics technique.

Possible code tweaks:
`PINImage+DecodedImage`
`PINProgressiveImage`

The tricky part here is that `FLAnimatedImage` doesn't do anything with wide color display, so information may got lost here. It may not be a big one, though, as .gifs are likely not the use case for images outside of the sRGB color space.

2) In addition, any `CGColorSpaceRef` instances would need to be set by using `CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3)` if the display supports it (i.e. `[[[UIScreen mainScreen] traitCollection] displayGamut] == UIDisplayGamutP3`) instead of the typical `CGColorSpaceCreateDeviceRGB`. This part seems to be a bit tricky, because we aren't really sure if the server is going to give us a wide color image, which means when downloading things in areas where we need color space information - I think we'd have to snag it by using `CGImageGetColorSpace`.

Possible code tweaks:
`PINAnimatedImage`
`PINImage+DecodedImage`
`PINImage+WebP`

3) Finally, in places where export the image back out, we'll want to replace
`UIImageJPEGRepresentation` with `UIGraphicsImageRenderer` and use its `jpegData` method.

Possible code tweaks:
`PINImage+DecodedImage`

Apologies for the brain dump, I'm relatively new to color space information, but I've tried to book up on it a bit and this what I came up with :). Instagram's post was super helpful too, which @ay8s linked to above. I'm definitely up to start hacking on this but wanted to see if any of this triggered thoughts on your end 👍

**Comment by Sumit1991Saha** [comment]
Hi, 
For the images to make wide color aware, we need to use something like mentioned below,
UIGraphicsImageRendererFormat *format =  [UIGraphicsImageRendererFormat new];
format.prefersExtendedRange = YES;
UIGraphicsImageRenderer *renderer = [[UIGraphicsImageRenderer alloc] initWithSize:size format:format];
//Do any manipulation

But to do so we need to write some initial bitmap image data captured thorugh camera via avcapturesession's object and write it down to the UIGraphicsImageRendererContext. I am stuck at that part, i am unable to do so.
Can you please help?
Thanks
Sumit

**Comment by IHEARTCOOKIES** [comment]
Should be addressed in #549 

cc: @garrettmoon @bolsinga
